frontend/src/components/Table/Table.tsx

```diff
@@ -234,25 +242,35 @@ const Table = <T extends object>(props: PropsWithChildren<TableProps<T>>): React
   }, [data, setSelectedRows, page, selectedRowIds]);
 
   const getNextSortDirection = (column: ColumnInstanceWithProps<T>): SortDirection => {
-    if (!props.sort) return 'asc';
+    if (!(props.sort as any)[column.id]) return 'asc';
 
-    if (props.sort.column !== column.id || props.sort.direction === 'desc') {
-      return 'asc';
+    if ((props.sort as any)[column.id] === 'desc') {
+      return undefined;
     }
 
     return 'desc';
   };
 
   const renderHeaderCell = (column: ColumnInstanceWithProps<T>) => {
-    const isSorted = props.sort && props.sort.column === column.id;
-    if (!column.sortable || !isSorted) {
-      return column.render('Header');
-    }
-
     return (
       <div className="sortable-column">
         {column.render('Header')}
-        {props.sort?.direction === 'asc' ? <FaLongArrowAltUp /> : <FaLongArrowAltDown />}
+        <ColumnSort
+          onSort={() => {
+            if (props.onSortChange && column.sortable) {
+            }
```

Comment:
are the brackets on this "if" right? it doesn't look like it's encapsulating anything

Reply:
Nice catch